### PR TITLE
[UIK-2348][button] fix disabled state for ButtonLink

### PR DESCRIPTION
**Pull request by ksimanov**
## Motivation and Context

 Fix disabled state wasn't applied to ButtonLink

## How has this been tested?

- manually
- I have added unit tests with screenshot

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly or it's not required.
- [x] Unit tests are not broken.
- [x] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [x] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

File: semcore/button/src/component/ButtonLink/ButtonLink.tsx [hunk 1]
```diff
@@ -12,6 +12,8 @@ const enhance = {
   resolveColor: resolveColorEnhance(),
 };
 
+// TODO: find how Button becames visually-disabled={disabled}
```
Comment (ilyabrower) [inline semcore/button/src/component/ButtonLink/ButtonLink.tsx:15]: My bad, `visually-disabled` comes from `Link` styles. Your fix is correct, but you don't need this comment

File: semcore/button/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [5.39.4] - 2024-10-30
+
+### Fixed
+
+- Fix disabled state is not applied to ButtonLink
```
Comment (ilyabrower) [inline semcore/button/CHANGELOG.md:9]: Let's write
```
### Fixed
- Disabled state....
```

File: semcore/button/src/component/ButtonLink/ButtonLink.tsx (hunk 1 above)
Comment (ksimanov) [reply semcore/button/src/component/ButtonLink/ButtonLink.tsx]: Fixed.

File: semcore/button/CHANGELOG.md (hunk 2 above)
Comment (ksimanov) [reply semcore/button/CHANGELOG.md]: Fixed.

File: semcore/button/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [5.39.4] - 2024-10-30
```
Comment (ilyabrower) [inline semcore/button/CHANGELOG.md:5]: why not 5.39.1 ?

Comment (ksimanov) [reply semcore/button/CHANGELOG.md]: You are right. Fixed.

File: semcore/button/CHANGELOG.md (hunk 2 above)
Comment (ksimanov) [reply semcore/button/CHANGELOG.md]: Now it's fixed for sure.

**Review by ilyabrower — APPROVED** [review]

**Review by Valeria-Zimnitskaya — APPROVED** [review]

File: semcore/button/src/component/ButtonLink/buttonLink.shadow.css [hunk 4]
```diff
@@ -57,7 +57,7 @@ SButton[keyboardFocused] {
   outline-width: 3px;
 }
 
-SButton[visually-disabled] {
+SButton[disabled] {
```
Comment (sheila-semrush) [inline semcore/button/src/component/ButtonLink/buttonLink.shadow.css:60]: epic fix))))

**Review by sheila-semrush — APPROVED** [review]
everything's good, I've also added one disabled button to the example

**Pull request merged**